store/store.go

```diff
@@ -531,11 +531,37 @@ func (s *SnapUbuntuStoreRepository) FindSnaps(searchTerm string, channel string,
 	return snaps, nil
 }
 
-// Updates returns the available updates for a list of snap identified by fullname with channel.
-func (s *SnapUbuntuStoreRepository) Updates(installed []string, auther Authenticator) (snaps []*snap.Info, err error) {
-	// XXX: uses obsolete end point!
+type metadataInput struct {
+	SnapID      string               `json:"snap_id"`
+	Channel     string               `json:"channel"`
+	Revision    int                  `json:"revision"`
+	Epoch       string               `json:"epoch"`
+	Confinement snap.ConfinementType `json:"confinement"`
+}
 
-	jsonData, err := json.Marshal(map[string][]string{"name": installed})
+type metadataWrapper struct {
+	Snaps  []metadataInput `json:"snaps"`
+	Fields []string        `json:"fields"`
+}
+
+// Updates returns the available updates for a list of snap identified by fullname with channel.
+func (s *SnapUbuntuStoreRepository) Updates(installed []*snap.Info, auther Authenticator) (snaps []*snap.Info, err error) {
```

Comment:
as we discussed in the standup this probably should take some variations of metadataInput (exported), with Info doesns't match exactly what we need and makes it tempting to take decisions here